src/Bookings.js

```diff
@@ -1,21 +1,125 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
+import ClipLoader from "react-spinners/ClipLoader";
 import Search from "./Search.js";
-// import SearchResults from "./SearchResults.js";
-// import FakeBookings from "./data/fakeBookings.json";
+import OutcomeSearch from "./OutcomeSearch.js";
+import CustomerProfile from "./CustomerProfile.js";
 
 const Bookings = () => {
-  const search = searchVal => {
-    console.info("TO DO!", searchVal);
+  const [bookings, setBookings] = useState([]);
+  const [customerId, setCustomerId] = useState(null);
+  const [data, setData] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [errorOccurred, setErrorOccurred] = useState({
+    ok: true,
+    status: "",
+    statusText: "",
+  });
+
+  const { ok, status, statusText } = errorOccurred;
+
+  const fetchBookings = async () => {
+    try {
+      const response = await fetch("https://cyf-react.glitch.me");
+      if (!response.ok) {
+        throw new Error(`${response.status} ${response.statusText}`);
+      }
+
+      const data = await response.json();
+      setBookings(data);
+      setData(data);
+      setIsLoading(false);
+      setErrorOccurred({ ok: true, status: "", statusText: "" });
+    } catch (error) {
+      setErrorOccurred({
+        ok: false,
+        status: error.toString(),
+        statusText: "Failed to fetch bookings data.",
+      });
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    fetchBookings();
+  }, []);
+
+  const search = (searchVal) => {
+    const trimmedSearchVal = searchVal.trim().toLowerCase();
+    if (trimmedSearchVal === "") {
+      setBookings(data);
+    } else {
+      const filterBookings = bookings.filter(
+        (element) =>
+          element.firstName.toLowerCase().startsWith(trimmedSearchVal) ||
+          element.surname.toLowerCase().startsWith(trimmedSearchVal)
+      );
+      setBookings(filterBookings);
+    }
   };
 
   return (
     <div className="App-content">
       <div className="container">
         <Search search={search} />
-        {/* <SearchResults results={FakeBookings} /> */}
+        {!ok ? (
+          <div className="loader">
+            <span>
+              An Error occurred when fetching the Bookings Data: {status}{" "}
+              {statusText}
+            </span>
+          </div>
+        ) : isLoading ? (
+          <div>
+            <ClipLoader color="green" size={70} />
+            <span>Loading, please wait...</span>
+          </div>
+        ) : (
+          <OutcomeSearch results={bookings} setCustomerId={setCustomerId} />
+        )}
+        {customerId && <CustomerProfile id={customerId} />}
```

Comment:
Is using the CustomerProfile component here needed? I see that you render the CustomerProfile component in the SearchResult.js line 54.

I think customerId is always undefined and thus it never renders the component here. Please take a look and let me know what are your thoughts on this. Thanks